pyomo/repn/standard_repn.py

```diff
@@ -734,16 +734,22 @@ def _collect_pow(exp, multiplier, idMap, compute_values, verbose, quadratic):
             elif compute_values and len(res.linear) == 0:
                 return Results(constant=multiplier*res.constant**exponent)
             #
-            # If there is one linear term, then we compute the quadratic expression for it.
+            # If the base is linear, then we compute the quadratic expression for it.
             #
-            elif len(res.linear) == 1:
-                key, coef = res.linear.popitem()
-                var = idMap[key]
+            else:
                 ans = Results()
-                if not (res.constant.__class__ in native_numeric_types and res.constant == 0):
+                has_constant = (res.constant.__class__
+                                not in native_numeric_types
+                                or res.constant != 0)
+                if has_constant:
                     ans.constant = multiplier*res.constant*res.constant
-                    ans.linear[key] = 2*multiplier*coef*res.constant
-                ans.quadratic[key,key] = multiplier*coef*coef
+                while len(res.linear) > 0:
+                    key1, coef1 = res.linear.popitem()
+                    if has_constant:
+                        ans.linear[key1] = 2*multiplier*coef1*res.constant
+                    ans.quadratic[key1,key1] = multiplier*coef1*coef1
+                    for key2, coef2 in res.linear.items():
+                        ans.quadratic[key1,key2] = 2*multiplier*coef1*coef2
```

Comment:
Note that this is inherently nondeterministic.  Unfortunately, this is a flaw in the current Results data structures.  It will be resolved n an upcoming rewrite of generate_standard_repn (see #1073 and the draft implementation on the [solver-rewrite ](https://github.com/jsiirola/pyomo/blob/solver-rewrite/pyomo/repn/new_standard_repn.py) branch)

Reply:
I stand corrected.  @carldlaird pointed out to me that this can be made deterministic relatively easily.  Expect a PR from him that will resolve this and then we can merge.